FinSearchUnified/Tests/Bundle/SearchBundleFindologic/FacetHandler/TextFacetHandlerTest.php

```diff
@@ -0,0 +1,360 @@
+<?php
+
+namespace FinSearchUnified\Tests\Bundle\SearchBundleFindologic\FacetHandler;
+
+use FinSearchUnified\Bundle\SearchBundleFindologic\FacetHandler\TextFacetHandler;
+use Shopware\Bundle\SearchBundle\Condition\ProductAttributeCondition;
+use Shopware\Bundle\SearchBundle\ConditionInterface;
+use Shopware\Bundle\SearchBundle\Criteria;
+use Shopware\Bundle\SearchBundle\Facet\ProductAttributeFacet;
+use Shopware\Bundle\SearchBundle\FacetResult\RadioFacetResult;
+use Shopware\Bundle\SearchBundle\FacetResult\ValueListFacetResult;
+use Shopware\Bundle\SearchBundle\FacetResult\ValueListItem;
+use Shopware\Components\Test\Plugin\TestCase;
+use SimpleXMLElement;
+
+class TextFacetHandlerTest extends TestCase
+{
+    /**
+     * @dataProvider filterProvider
+     *
+     * @param string $name
+     * @param string $type
+     * @param bool $doesSupport
+     */
+    public function testSupportsFilter($name, $type, $doesSupport)
+    {
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+
+        $filter->addChild('name', $name);
+        $filter->addChild('type', $type);
+
+        $facetHandler = new TextFacetHandler();
+        $result = $facetHandler->supportsFilter($filter);
+
+        $this->assertSame($doesSupport, $result);
+    }
+
+    public function filterProvider()
+    {
+        return [
+            'Category filter with "select" type' => ['cat', 'select', false],
+            'Category filter with "label" type' => ['cat', 'label', false],
+            'Vendor filter with "select" type' => ['vendor', 'label', true],
+            'Vendor filter with "label" type' => ['vendor', 'label', true],
+        ];
+    }
+
+    public function testFacetModeForBooleanResult()
+    {
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+
+        $facet = new ProductAttributeFacet(
+            'field',
+            ProductAttributeFacet::MODE_BOOLEAN_RESULT,
+            'field',
+            'Field'
+        );
+
+        $facetHandler = new TextFacetHandler();
+        $result = $facetHandler->generatePartialFacet($facet, new Criteria(), $filter);
+
+        $this->assertNull($result);
+    }
+
+    /**
+     * @dataProvider valueListDataProvider
+     *
+     * @param array $filterData
+     * @param ValueListFacetResult $facetResult
+     * @param ConditionInterface|null $condition
+     */
+    public function testFacetModeForListFacetResult(
```

Comment:
I think `testFacetModeForListFacetResult` and `testFacetModeForRadioFacetResult` don't have to be separate tests since they are 99% similar. My suggestion would be:
1. Rename the test to `testGeneratesPartialFacetBasedOnFilterDataAndActiveConditions`.
2. Use `FacetResultInterface` as typehint instead of `ValueListFacetResult` or `RadioFacetResult`.
3. Use `valueListDataProvider` and `radioListDataProvider` for the same test.
4. Pass the mode via the data provider instead of hard coding it.

At the end you should only have `testGeneratesPartialFacetBasedOnFilterDataAndActiveConditions` instead of `testFacetModeForListFacetResult` and `testFacetModeForRadioFacetResult`.